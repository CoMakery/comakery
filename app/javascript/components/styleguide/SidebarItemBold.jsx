import React from 'react'
import classNames from 'classnames'
import SidebarItem from './SidebarItem'

class SidebarItemBold extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'sidebar-item__bold',
      className
    )

    return (
      <React.Fragment>
        <SidebarItem className={classnames} {...other} />
      </React.Fragment>
    )
  }
}

export default SidebarItemBold
