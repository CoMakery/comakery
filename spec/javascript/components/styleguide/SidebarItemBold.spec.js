import React from 'react'
import { shallow, mount, render } from 'enzyme'
import SidebarItemBold from 'components/styleguide/SidebarItemBold'

describe('SidebarItemBold', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<SidebarItemBold/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.exists('.sidebar-item__bold')).toBe(true)
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<SidebarItemBold className='__test' />)

      expect(wrapper.exists('.sidebar-item__bold.__test')).toBe(true)
    })
})
