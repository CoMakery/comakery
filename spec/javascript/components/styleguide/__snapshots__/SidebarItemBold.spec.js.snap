// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItemBold renders correctly without props 1`] = `
<Fragment>
  <SidebarItem
    className="sidebar-item__bold"
    iconLeftName=""
    iconLeftUrl=""
    iconRightName=""
    notificationColor=""
    selected={false}
    text="sidebar item"
  />
</Fragment>
`;
